### PR TITLE
Adapt to new select target restrictions

### PR DESCRIPTION
**Pull request by mfelsche**
Adapt docs to new restrictions for the select target expression introduced in: https://github.com/tremor-rs/tremor-runtime/pull/828

### CODE REVIEW

File: docs/tremor-query/index.md [hunk 1]
```diff
@@ -120,6 +120,13 @@ Tremor supports tumbling windows by number of events or by time.
 General configuration Parameters:
 
 * `eviction_period`: duration in nanoseconds without events arriving, after which to evict / remove the current window data for a single group.
+* `max_groups`: maximum number of groups to maintain simultaneously in memory. Groups added beyond that number will be ignored. Per default, tremor allows as much groups as will fit into memory.
```
Comment (Licenser) [inline docs/tremor-query/index.md:123]: I think this needs re-wording, by default tremor does not limit the number of groups if a user runs out of memory it swaps and then eventually crashes with an out of memory error

File: docs/tremor-query/index.md [hunk 2]
```diff
@@ -120,6 +120,13 @@ Tremor supports tumbling windows by number of events or by time.
...
+
+Each select statement maintains the groups for the current windows in an in memory data-structure. This contains the group values as well as the aggregate states.
+If your grouping values possibly have a very high cardinality it is possible to end up with runaway memory growth, as per default the group data structures won't be evicted,
+unless `eviction_period` is set. Old groups will be discarded after `2 x eviction_period` if no event for those groups arrived.
```
Comment (Licenser) [inline docs/tremor-query/index.md:127]: Is this correct? I was under the impression that they get evicted as well when they emit unless `emit_empty_windows` is selected.

File: docs/tremor-query/index.md [hunk 3]
```diff
@@ -146,6 +153,7 @@ independent from event flow with a granularity of `100ms`. It is thus possible t
 Configuration Parameters:
 
 - `interval`: Time interval in nanoseconds after which the window closes.
+- `emit_empty_windows` - By default, time based windows will only emit, if events arrived. By configuring `emit_empty_windows` as `true` this window will emit every `interval`, regardless if events arrived or not. If you use this in a `group by` query and the cardinality is likely huge, consider using `max_groups` and `eviction_period` to avoid runaway memory growth such a window will one event per interval and group for which we've seen events before.
```
Comment (Licenser) [inline docs/tremor-query/index.md:156]: We should add a `!warning` callout here

**Review by Licenser — CHANGES_REQUESTED** [review]
few minor comments, but generally :+1: , and nice clean up on the workshops!

File: docs/tremor-query/index.md (hunk 2 above)
Comment (mfelsche) [reply docs/tremor-query/index.md:127]: when they emit, the aggregate states get reset, but the groups remain in the internal map inside the window.

I think we might need to spend some time coming up with a better default behaviour that cleans up as quickly as possible (in an LRU manner maybe).

Comment (Licenser) [reply docs/tremor-query/index.md:127]: Ja this sounds totally wrong to me IMHO, let's make a ticket and clean that up, I see horrible failure cases in here :joy: (but not related to the doc PR so a nonissue here)

**Review by Licenser — APPROVED** [review]
Nice :+1: thank you!

**Pull request merged**